lib/metrics.js

```diff
@@ -15,67 +28,43 @@ module.exports = {
 function prepareData(res) {
   const audits = res.audits;
 
-  const resFMP = audits['first-meaningful-paint'];
-  const resFMPext = resFMP.extendedInfo;
-  const resSI = audits['speed-index-metric'];
-  const resSIext = resSI.extendedInfo;
-  const resTTI = audits['time-to-interactive'];
-  const resTTIext = resTTI.extendedInfo;
-
   const colorP0 = 'yellow';
   const colorP2 = 'green';
   const colorVisual = 'blue';
 
-  const timestamps = [{
-    title: 'Navigation Start',
-    name: 'navstart',
-    value: resFMPext && resFMPext.value.timestamps.navStart
-  }];
+  const timings = [];
+  metricsDefinitions.forEach(metric => {
+    try {
+      const resolvedMetric = {
+        title: metric.name,
+        name: metric.id,
+        value: metric.getTs(audits)
```

Comment:
The reason of using metrics definition here is reducing [prepareData](https://github.com/paulirish/pwmetrics/blob/master/lib/metrics.js#L15). 
If we take look at e.g. `tti` results, it has `timings` and `timestamps` 
```
 "time-to-interactive": {
        "score": 100,
        "displayValue": "526.5ms",
        "rawValue": 526.5,
        "optimalValue": "5,000ms",
        "extendedInfo": {
            "value": {
                "timings": {
                    "fMP": 508.2,
                    "visuallyReady": 526.508,
                    "timeToInteractive": 526.508
                },
                "timestamps": {
                    "fMP": 114185594921,
                    "visuallyReady": 114185613239,
                    "timeToInteractive": 114185613239
                },
                "expectedLatencyAtTTI": 16,
                "foundLatencies": [
                    {
                        "estLatency": 16,
                        "startTime": "526.5"
                    }
                ]
            },
            "formatter": "null"
        },
        "name": "time-to-interactive",
        "category": "Performance",
        "description": "Time To Interactive (alpha)",
        "helpText": "Time to Interactive identifies the time at which your app appears to be ready enough to interact with. [Learn more](https://developers.google.com/web/tools/lighthouse/audits/time-to-interactive)."
    },
```

Right now we can operate this data as we wish, but using [API](https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-core/lib/traces/pwmetrics-events.js#L32) we can't reach `timings` because it provide us with `timestamps`, `name` and `id`.
What I see here, we can either extend lighthouse API with adding one more method `getTimings` or leave it as it is right now...